static/src/javascripts/projects/common/modules/commercial/creatives/hosted-thrasher.js

```diff
@@ -0,0 +1,35 @@
+define([
+    'common/utils/fastdom-promise',
+    'common/utils/template',
+    'common/modules/commercial/hosted-video',
+    'common/modules/commercial/creatives/add-tracking-pixel',
+    'text!common/views/commercial/creatives/hosted-thrasher.html'
+], function (
+    fastdom,
+    template,
+    hostedVideo,
+    addTrackingPixel,
+    hostedThrasherStr
+) {
+    var hostedThrasherTemplate;
+
+    var HostedThrasher = function ($adSlot, params) {
+        this.$adSlot = $adSlot;
+        this.params = params;
+    };
+
+    HostedThrasher.prototype.create = function () {
+        hostedThrasherTemplate = hostedThrasherTemplate || template(hostedThrasherStr);
+
+        fastdom.write(function () {
+            this.$adSlot.append(hostedThrasherTemplate({ data: this.params}));
+        }, this).then(hostedVideo.init);
+
+        if (this.params.trackingPixel) {
```

Comment:
Maybe do this in the fastdom call, too?

Other than that, though, it all looks good.